### PR TITLE
Migration to CircleCI 2.0

### PR DESCRIPTION
**Pull request by LindsayHill**
Converting CircleCI config to 2.0 style.

This could be *massively* improved, using docker rather than machine executor type. 
The st2box test images need to be changed too - they are running 1.6dev, so tests are fully valid.
We also need to check for errors in the test log output, e.g. `ERROR Stream error: Event { type: 'error', status: 404 }`

Still, overall improvement here: Build times around 6:30, well down from 13-20minutes.

The `deploy:` job has not yet been tested - this can only be done on merge to master.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          name: Install dependencies
+          command: |
+            set -x
+            sudo apt-get -y install parallel
+            gem install package_cloud
+      - run:
+          name: Deploy deb/rpm packages to PackageCloud
+          command: |
+            "parallel -v -j0 --line-buffer  ~/st2-packages/.circle/packagecloud.sh deploy {} ~/packages/{} ::: ${DISTROS}"
+            ~/st2-packages/.circle/save_payload.py ~/packages
```
Comment (arm4b) [inline .circleci/config.yml]: This line could be removed, not used/needed anymore.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      ST2_PASSWORD: 123
+    steps:
+    - checkout
+    - run:
+        name: Create directories and get repos and dependencies
+        command: |
+          mkdir -p ~/packages /tmp/st2chatops/log
+          git clone --depth 1 ${ST2_PACKAGES_REPO} ~/st2-packages
+          git clone --depth 1 ${ST2_TEST_ENVIRONMENT} ~/st2box
+          sudo apt-get update -qq && sudo apt-get install -y rpm parallel jq
```
Comment (arm4b) [inline .circleci/config.yml]: `parallel` package is not needed in this step, as it's used in deploy section

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      - build
+      - deploy:
+          requires:
+            - build
+          filters:
+            branches:
+              only:
+                - master
+                - /v[0-9]+\.[0-9]+/
+                - feature/circleci
```
Comment (arm4b) [inline .circleci/config.yml:140]: @LindsayHill You can temporary add `circle2.0` branch here to test the deployment step so there won't be surprises during the upcoming release.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
+version: 2
```
Comment (arm4b) [inline .circleci/config.yml:7]: Please move the code comments from the previous circle config starting with
>  `# Setup in CircleCI account the following ENV variables: `

Might save time for someone as it's not clear at all which env variables are required for this build to work.

**Comment by LindsayHill** [comment]
@armab OK, I think I've addressed everything, and I've done deployment tests. All looks OK. Should be good to merge now?

**Review by arm4b — APPROVED** [review]
LGTM :+1:

**Pull request merged**